test/functional/p2p_invalid_messages.py

```
@@ -4,10 +4,14 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test node responses to invalid network messages."""
 
+import random
+import struct
```

Ruff (0.12.2)

`struct` imported but unused

Remove unused import: `struct`

(F401)